### PR TITLE
Automatically sync a Slack channel on webhook "channel joined"

### PR DESCRIPTION
**Pull request by lasryaric**

### CODE REVIEW

File: connectors/src/api/webhooks/webhookSlack.ts [hunk 1]
```diff
@@ -83,14 +93,31 @@ const _webhookSlackAPIHandler = async (
...
+    } else if (req.body.event?.type === "member_joined_channel") {
+      if (!req.body.event?.channel || !req.body.event?.user) {
+        return res.status(400).send({
+          error: {
+            message:
+              "Missing channel or user in request body for member_joined_channel event",
+          },
+        });
+      }
+      await launchSlackUserJoinedWorkflow(
```
Comment (spolu) [inline connectors/src/api/webhooks/webhookSlack.ts:105]: Result is not unpacked

File: connectors/src/connectors/slack/temporal/workflows.ts [hunk 2]
```diff
@@ -228,6 +229,37 @@ export async function syncOneMessageDebounced(
...
+export async function memberJoinedChannel(
+  connectorId: string,
+  nangoConnectionId: string,
+  dataSourceConfig: DataSourceConfig,
+  channelId: string,
+  userId: string
+): Promise<void> {
+  const slackAccessToken = await getAccessToken(nangoConnectionId);
+  const myUserId = await whoAmI(slackAccessToken);
+  if (myUserId == userId) {
```
Comment (spolu) [inline connectors/src/connectors/slack/temporal/workflows.ts:241]: Shouldn't this be tested before we start the workflow?

**Review by spolu — COMMENTED** [review]
Left a question (don't we want to test whether the update is about us before we launch the workflow vs in the workflow itself, otherwise we'll be launching workflows for nothing quite frequently which is not great from an observability standpoint)?

File: connectors/src/connectors/slack/temporal/workflows.ts (hunk 2 above)
Comment (lasryaric) [reply connectors/src/connectors/slack/temporal/workflows.ts:241]: Yes you are right, I just wanted to keep the strucutre of only talking to slack in Temporal, but, I will fix that.

**Review by spolu — APPROVED** [review]
LGTM

**Pull request merged**